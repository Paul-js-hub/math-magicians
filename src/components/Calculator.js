import React, { Component } from 'react';
// import Button from './Button';
import calculate from '../logic/calculate';
import './calculator.css';

export default class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      array: ['AC', '+/-', '%', 7, 8, 9, 4, 5, 6, 1, 2, 3, 0, '.'],
      total: 0,
      next: '',
      operation: '',
    };
  }

  handleClick = (obj) => {
    const value = obj.target.textContent;
    this.setState((e) => calculate(e, value));
  }

  render() {
    const {
      array, total, next, operation,
    } = this.state;
    const result = `${total || operation || next ? `${total || ''} ${operation || ''} ${next || ''}` : 0}`;
    return (
      <div className="container">
        <div className="input">{result}</div>
        <div className="array-btn flex">
          {array.map((arr) => <button type="button" key={arr} onClick={this.handleClick} className="btn-array">{arr}</button>)}
        </div>
        <div className="operations subgrid">
          <button type="button" onClick={this.handleClick} className="opers">รท</button>
          <button type="button" onClick={this.handleClick} className="opers">x</button>
          <button type="button" onClick={this.handleClick} className="opers">-</button>
          <button type="button" onClick={this.handleClick} className="opers">+</button>
          <button type="button" onClick={this.handleClick} className="opers">=</button>
        </div>
      </div>
    );
  }
}
