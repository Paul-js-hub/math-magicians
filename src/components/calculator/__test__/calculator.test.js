import { render, screen, fireEvent} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

test('it returns a calculator component', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it expects a particular button to be clicked', () => {
  const handleClick = jest.fn();
  render(<button onClick={handleClick}>x</button>)
  fireEvent.click(screen.getByRole('button', { name: /x/i }));
  expect(handleClick).toHaveBeenCalledTimes(1);
});
