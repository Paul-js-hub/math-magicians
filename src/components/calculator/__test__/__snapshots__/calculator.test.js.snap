// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it returns a calculator component 1`] = `
<div
  className="container"
>
  <div
    className="input"
  >
    0
  </div>
  <div
    className="array-btn flex"
  >
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn-array"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
  </div>
  <div
    className="operations subgrid"
  >
    <button
      className="opers"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="opers"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="opers"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="opers"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="opers"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
